### PR TITLE
GEODE-3791: add new tests for CacheListener on PartionedRegion

### PR DESCRIPTION
**Pull request by kirklund**
Note: We also changed the getOldValue assertions in the base test to match the javadocs which say that it may return the old value or null.

### CODE REVIEW

**Comment by dschneider-pivotal** [comment]
You need to run spotlessApply

File: geode-core/src/test/java/org/apache/geode/internal/cache/PRCacheListenerInvocationTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.test.junit.categories.DistributedTest;
+
+/**
+ * This class tests event triggering and handling in partitioned regions.
+ *
+ * <p>
+ * Converted from JUnit 3.
+ *
+ * @since 5.1
```
Comment (dschneider-pivotal) [inline geode-core/src/test/java/org/apache/geode/internal/cache/PRCacheListenerInvocationTest.java]: Did you intend to leave these since tags in?

Comment (kirklund) [reply geode-core/src/test/java/org/apache/geode/internal/cache/PRCacheListenerInvocationTest.java]: Added GemFire to the `@since` tags.

**Comment by kirklund** [comment]
@dschneider-pivotal fixed the spotless format and add GemFire to the `@since` tags.
